xdsl/dialects/comb.py

```diff
@@ -153,21 +227,115 @@ class XorOp(VariadicCombOperation):
     name = "comb.xor"
 
 
+@dataclass
+class ComparisonOperation:
```

Comment:
I'd move this base class up under `BinCombOperation`.

Reply:
Also, I think this should be 
```suggestion
class ComparisonOperation(IRDLOperation, ABC):
```

while removing the `IRDLOperation` inheritance from the classes that inherit from it below.